### PR TITLE
DEV-1692: Upgrade linter and fix issues

### PR DESCRIPTION
**Pull request by m-terel**

### CODE REVIEW

File: cmd/rportd/main.go [hunk 1]
```diff
@@ -225,8 +225,8 @@ func runMain(*cobra.Command, []string) {
 	}
 
 	// create --data-dir path if not exist
-	if err := os.MkdirAll(cfg.DataDir, os.ModePerm); err != nil {
-		log.Printf("ERROR: failed to create --data-dir %q: %v\n", cfg.DataDir, err)
+	if makedirErr := os.MkdirAll(cfg.DataDir, os.ModePerm); makedirErr != nil {
```
Comment (martinlindhe) [inline cmd/rportd/main.go:228]: It doesnt look like `err` being shadowed, what is the purpose of this?

File: server/sessions/data_test.go [hunk 2]
```diff
@@ -191,3 +191,24 @@ var s4 = &ClientSession{
 	Tunnels:      make([]*Tunnel, 0),
 	Disconnected: &s4DisconnectedTime,
 }
+
+// shallowCopy is used only in tests.
```
Comment (martinlindhe) [inline server/sessions/data_test.go:195]: // shallowCopy is used in tests to avoid copying lock value of mutex

Comment (m-terel) [reply server/sessions/data_test.go:195]: Updated.

File: cmd/rportd/main.go (hunk 1 above)
Comment (martinlindhe) [reply cmd/rportd/main.go:228]: Gotcha. (I wonder why I'm not seeing this error locally)

**Review by martinlindhe — APPROVED** [review]

**Pull request merged**